src/ln/channelmanager.rs

```diff
@@ -661,12 +661,12 @@ impl ChannelManager {
 
 	/// Call this upon creation of a funding transaction for the given channel.
 	/// Panics if a funding transaction has already been provided for this channel.
-	pub fn funding_transaction_generated(&self, temporary_channel_id: &Uint256, funding_txo: (Sha256dHash, u16)) {
+	pub fn funding_transaction_generated(&self, temporary_channel_id: &Uint256, funding_txo: TxOutRef) {
```

Comment:
Ugh, I dont like this. Practically no one should ever create a funding transaction (or any transaction) with more than 2**16 outputs, but its not technically impossible, so letting the user provide a usize instead of a u16 feels icky cause we're going a down-convert without making the user aware of it.